src/event.bif

```diff
@@ -885,8 +885,12 @@ event Pcap::file_done%(path: string%);
 ##
 ## first_bytes: A certain number of bytes at the start of the unknown protocol's header.
 ##
+## analyzer_history: The chain of packet analyzers that processed the packet up to this
+##                   point. This includes the history of encapsulating packets in case
+##                   of tunneling.
+##
 ## .. zeek:see:: UnknownProtocol::first_bytes_count
-event unknown_protocol%(analyzer_name: string, protocol: count, first_bytes: string%);
+event unknown_protocol%(analyzer_name: string, protocol: count, first_bytes: string, analyzer_history: string_vec%);
```

Comment:
Two thoughts:
* We need to keep the old `unknown_protocol()` event and deprecate it for 8.1 (see recent commit 401a07403628f32c09eb03c57dba5e93fa35681b). Even if you're the only user ;-)
* A bit nit-picky, wondering about returning `vector of PacketAnalyzer::Tag` rather than the string version. The tag seems nicer API wise. For getting a short history as you have right now, this needs fiddling with the stringified tag in script land unfortunately, so I can certainly see the convenience if you want to push back on that idea.

Reply:
Ah, I'm seeing `unknown_protocol()` already gets an analyzer name instead of a tag. Ignore point two and lets stick with a string for consistency.

Reply:
I've added deprecation for the old event signature.

FWIW, my gut feeling was that the tag version could also come with a measurable performance hit because the Tags need to be translated in script-land. I've [implemented it](https://github.com/J-Gras/zeek/tree/topic/jgras/packet-analyzer-history-rework-tags) but couldn't see any significant impact for the given workload in my local tests:
```
$ hyperfine -w 1 -m 20 "~/zeek-installs/zeek-7.1.0-analyzer-history-rework-tags/bin/zeek -CD -r 2009-M57-day11-18.pcap" "~/zeek-installs/zeek-7.1.0-analyzer-history-rework-raw/bin/zeek -CD -r 2009-M57-day11-18.pcap"
Benchmark 1: ~/zeek-installs/zeek-7.1.0-analyzer-history-rework-tags/bin/zeek -CD -r 2009-M57-day11-18.pcap
  Time (mean ± σ):      2.707 s ±  0.044 s    [User: 2.208 s, System: 1.985 s]
  Range (min … max):    2.660 s …  2.860 s    20 runs
 
Benchmark 2: ~/zeek-installs/zeek-7.1.0-analyzer-history-rework-raw/bin/zeek -CD -r 2009-M57-day11-18.pcap
  Time (mean ± σ):      2.698 s ±  0.065 s    [User: 2.205 s, System: 1.978 s]
  Range (min … max):    2.584 s …  2.874 s    20 runs
 
Summary
  ~/zeek-installs/zeek-7.1.0-analyzer-history-rework-raw/bin/zeek -CD -r 2009-M57-day11-18.pcap ran
    1.00 ± 0.03 times faster than ~/zeek-installs/zeek-7.1.0-analyzer-history-rework-tags/bin/zeek -CD -r 2009-M57-day11-18.pcap
```